src/extra/components/dld.py

```diff
@@ -0,0 +1,304 @@
+
+from copy import copy
+import re
+
+import numpy as np
+import pandas as pd
+
+
+class DelayLineDetector:
+    """Interface for processed delay line detector data.
+
+    Args:
+        data (extra_data.DataCollection): Data to access DLD data from.
+        detector (str, optional): Source name of the detector, only
+            needed if the data includes more than one.
+        pulses (extra.components.pulses.PulsePattern, optional): Pulse
+            component to pull pulse information. If omitted, it is
+            constructed from the internal trigger information.
+    """
+
+    _instrument_re = re.compile(r'^(\w{3}_\w+_DLD\d\/DET\/\w+):output?$')
+
+    def __init__(self, data, detector=None, pulses=None):
+        if detector is None:
+            # Try to find detector automatically.
+            detector = self._find_detector(data)
+        elif detector.endswith(':output'):
+            # Strip output pipeline if passed
+            detector = detector[:-7]
+
+        self._detector_name = detector
+
+        if detector in data.control_sources:
+            # Some very early data may not have the control source yet.
+            self._control_src = data[detector]
+        else:
+            self._control_src = None
+
+        self._instrument_src = data[f'{detector}:output']
+        self._pulses = pulses
+
+        if self._pulses is None:
+            self._pulses = self.pulses()
+
+    def __repr__(self):
+        return "<{} {}>".format(
+            type(self).__name__, self._detector_name)
+
+    @classmethod
+    def _find_detector(cls, data):
+        """Try to find detector source."""
+
+        detectors = set()
+
+        for source in data.instrument_sources:
+            m = cls._instrument_re.match(source)
```

Comment:
For sources coming from Karabo, we're trying to use the device class, with the instrument pattern just as a fallback. What do you think about doing something similar here, introducing some kind of 'source type' metadata for the sources we create in calibration?

Obviously this doesn't need to be part of this PR - I'm thinking about the future.

Reply:
Yes definitely, I omitted it here as essentially all data is legacy data for now.

In an upcoming PR on another component, I started abstracting all this search-sources-based-on-device-class-or-needle functionality.